RecoVertex/BeamSpotProducer/src/PVFitter.cc

```diff
@@ -102,52 +92,38 @@ PVFitter::~PVFitter() {
 
 void PVFitter::readEvent(const edm::Event& iEvent)
 {
-
-//   frun = iEvent.id().run();
-//   const edm::TimeValue_t ftimestamp = iEvent.time().value();
-//   const std::time_t ftmptime = ftimestamp >> 32;
-
-//   if (fbeginLumiOfFit == -1) freftime[0] = freftime[1] = ftmptime;
-//   if (freftime[0] == 0 || ftmptime < freftime[0]) freftime[0] = ftmptime;
-//   const char* fbeginTime = formatTime(freftime[0]);
-//   sprintf(fbeginTimeOfFit,"%s",fbeginTime);
-
-//   if (freftime[1] == 0 || ftmptime > freftime[1]) freftime[1] = ftmptime;
-//   const char* fendTime = formatTime(freftime[1]);
-//   sprintf(fendTimeOfFit,"%s",fendTime);
-
-//   flumi = iEvent.luminosityBlock();
-//   frunFit = frun;
-
-//   if (fbeginLumiOfFit == -1 || fbeginLumiOfFit > flumi) fbeginLumiOfFit = flumi;
-//   if (fendLumiOfFit == -1 || fendLumiOfFit < flumi) fendLumiOfFit = flumi;
-//   std::cout << "flumi = " <<flumi<<"; fbeginLumiOfFit = " << fbeginLumiOfFit <<"; fendLumiOfFit = "<<fendLumiOfFit<<std::endl;
-
   //------ Primary Vertices
   edm::Handle< reco::VertexCollection > PVCollection;
   bool hasPVs = false;
-  //edm::View<reco::Vertex> vertices;
-  //const reco::VertexCollection & vertices = 0;
 
   if ( iEvent.getByToken(vertexToken_, PVCollection ) ) {
-      //pv = *PVCollection;
-      //vertices = *PVCollection;
       hasPVs = true;
   }
   //------
 
   if ( hasPVs ) {
 
       for (reco::VertexCollection::const_iterator pv = PVCollection->begin(); pv != PVCollection->end(); ++pv ) {
-
-
-           //for ( size_t ipv=0; ipv != pv.size(); ++ipv ) {
+          if (useOnlyFirstPV_){
+             if (pv != PVCollection->begin()) break;
+          }
 
           //--- vertex selection
           if ( pv->isFake() || pv->tracksSize()==0 )  continue;
           if ( pv->ndof() < minVtxNdf_ || (pv->ndof()+3.)/pv->tracksSize()<2*minVtxWgt_ )  continue;
           //---
 
+          if (pv->tracksSize() < minVtxTracks_ ) continue;
+		  
+          double sumPt=0;
+          for(auto iTrack = pv->tracks_begin(); iTrack != pv->tracks_end(); ++iTrack)
```

Comment:
Following https://github.com/cms-sw/cmssw/pull/18201#issuecomment-295599218 : indeed, I have the impression that either the relevant Track collection is added to the ALCARECO event content, or a further switch should be added here in the code, that allows the calculation of sumPt only when it is "true" (and it has to be kept "false" in the configurations meant to the AlCa workflows).
@francescobrivio : can you quickly have a look?